### DISCUSSION TITLE
How may I retrieve seq show-key when running seq in a container instance

### DISCUSSION THREAD

**Discussion by rodneytamblyn** [datalust/seq-tickets#2166] [Help] [answered]
Seq Settings > Backup has a notice:
```
Without a backup of the Seq secret key, backup files cannot be restored. The key must be retrieved in cleartext using the seq show-key command
```

I am running Seq in an Azure container instance, and don't have access to the seq command line (AFAIK).  How can I retrieve this key?

Rod

**Comment by KodrAus** [comment]
Hi @rodneytamblyn :wave:

There are two main ways you can grab your secret key from a docker container.

1. If you update your Azure Container Instance configuration to pass the `show-key` argument to your container and restart it, it'll print the key to stdout, which you should be able to find through the Azure UI. This is the equivalent to `docker run datalust/seq:latest show-key`.
2. If you're able to run a command through Azure on your running container, you can also run `seqsvr show-key`. This is the equivalent to `docker exec <your-seq-container> seqsvr show-key`.

Hope that helps!

**Reply by rodneytamblyn** [reply]
Assuming I am creating from az command line with something like:

```
 az container create --resource-group MYGROUP --file config.yaml --query seqsvr show-key
```
... results in --query: invalid jmespath_type value: 'seqsvr show-key'

Can you explain how to issue the command either via az container create or via parameter in the YAML file?

**Reply by nblumhardt** [reply, marked as answer]
Hi Rodney, thanks for the update. There's a simpler way to grab the key when running under Docker/ACI - in your mounted `/data` volume, you'll find the `Seq.json` file - you can grab the secret key from there, it's in `storage.secretKey`. Let me know if this does the trick.
